CONTRIBUTING.md

```diff
@@ -1,31 +1,66 @@
+
+Do not add your issue as a comment to an existing issue unless it's for the identical input. Many issues look similar, but have different causes.
+
+The more information you can provide, the more likely someone will be successful at reproducing the issue and finding a fix.
+
+Please include the following with each issue:
+
+- Version of JavaScript SDK
+
+- JavaScript platform and version you are using to develop your application. An example would be Angular 8.0.0.
+
+- Reproducible steps (1... 2... 3...) that cause the issue
+
+- What you expected to see, versus what you actually saw
+
+- Images, animations, or a link to a video showing the issue occurring
 
-- For faster response, we recommend using [Kinvey support](https://support.kinvey.com/support/home).
-- For bugs and enhancements specific to the library that are not time critical, please [file a GitHub issue](https://github.com/Kinvey/js-sdk/issues).
+- A code snippet that demonstrates the issue or a link to a code repository the developers can easily pull down to recreate the issue locally
 
-### Filing a bug
+  - **Note:** Because the developers need to copy and paste the code snippet, including a code snippet as a media file (i.e. .gif) is not sufficient.
 
-When filing a bug, please provide as much of the following information as possible -
+### Final Checklist
 
-- Steps to reproduce
-- Expected result
-- Actual result
-- Sample code that demonstrates the issue
-- Version of Kinvey / Device and toolset information
-- Kinvey app ID (**never post your app secrets publicly**)
+Please remember to do the following:
 
+- [ ] Search the issues to ensure your report is a new issue
 
-## Contributing Code
+- [ ] Simplify your code around the issue to better isolate the problem
 
-When contributing to the repository, please follow these guidelines -
+Don't feel bad if the developers can't reproduce the issue right away. They will simply ask for more information!
 
-- All contributions must be in the form of a pull request, with the `master` branch as the base.
-- Pull requests must pass the unit tests in the codebase. Whenever possible, new code should also add unit tests.
-- Each commit should include a commit message describing the change; and each pull request should include a high level summary of the bug or enhancement being addressed by the pull request.
-- The core team (Kinvey engineering) is currently responsible for reviewing, merging and releasing new code.
+## Contributing Fixes
 
-We require that all contributors sign a [CLA](https://goo.gl/forms/spZb2rXhC6I6zOxw1) prior to making a contribution. Please fill out the CLA so we can accept your contributions. If you have questions, you can email mobile@kinvey.com.
+If you are interested in writing code to fix issues, please see [How to Contribute](https://github.com/Kinvey/js-sdk/wiki/How-to-Contribute) in the wiki
```

Comment:
I like a lot of the changes here, but let's discuss them as a team, because I would think that we can pretty much have the same contribution guidelines across SDKs (with possibly a few per-platform additions).